cmd/alias.go

```diff
@@ -0,0 +1,153 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"github.com/fatih/color"
+	"github.com/posener/complete"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/mitchellh/cli"
+	"trellis-cli/trellis"
+)
+
+type AliasCommand struct {
+	UI      cli.Ui
+	flags   *flag.FlagSet
+	Trellis *trellis.Trellis
+	local   string
+}
+
+func NewAliasCommand(ui cli.Ui, trellis *trellis.Trellis) *AliasCommand {
+	c := &AliasCommand{UI: ui, Trellis: trellis}
+	c.init()
+	return c
+}
+
+func (c *AliasCommand) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flags.Usage = func() { c.UI.Info(c.Help()) }
+	c.flags.StringVar(&c.local, "local", "development", "local environment name, default: development")
+}
+
+func (c *AliasCommand) Run(args []string) int {
+	if err := c.Trellis.LoadProject(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+
+	args = c.flags.Args()
+
+	switch len(args) {
+	case 0:
+	default:
+		c.UI.Error(fmt.Sprintf("Error: too many arguments (expected 0, got %d)\n", len(args)))
+		c.UI.Output(c.Help())
+		return 1
+	}
+
+	// Copy files from package to Trellis
+	files := []string{"alias.yml", "alias.yml.j2", "alias-copy.yml"}
+	for _, file := range files {
+		_, filename, _, _ := runtime.Caller(0)
+		source := path.Join(path.Dir(filename), "../playbooks/" + file)
+		copyPlaybook(source, file)
+		defer deletePlaybook(file)
+	}
+
+	environments := c.Trellis.EnvironmentNames()
+	var remoteEnvironments []string
+	for _, environment := range environments {
+		if environment != c.local {
+			remoteEnvironments = append(remoteEnvironments, environment)
+		}
+	}
+
+	tempDir, tempDirErr := ioutil.TempDir("", "trellis-alias-")
+	if tempDirErr != nil {
+		log.Fatal(tempDirErr)
+	}
+	defer os.RemoveAll(tempDir)
+
+	for _, environment := range remoteEnvironments {
+		alias := execCommand("ansible-playbook", "alias.yml", "-vvv","-e", "env=" + environment, "-e", "trellis_alias_j2=alias.yml.j2", "-e", "trellis_alias_temp_dir=" + tempDir)
+		appendEnvironmentVariable(alias, "ANSIBLE_RETRY_FILES_ENABLED=false")
+
+		logCmd(alias, c.UI, true)
+		err := alias.Run()
+
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error running ansible-playbook alias.yml: %s", err))
+			return 1
+		}
+	}
+
+	combined := ""
+	for _, environment := range remoteEnvironments {
+		part, err := ioutil.ReadFile(tempDir + "/" + environment + ".yml.part")
```

Comment:
It might matter for Windows (assuming we support it properly), but we should probably always use `filepath.Join` for proper separators

Reply:
good catch